nixnet/database/_dbc_attributes.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import six
+import typing  # NOQA: F401
+
+from nixnet import _funcs
+from nixnet import constants
+
+
+class DbcAttributeCollection(collections.Mapping):
+    """Collection for accessing DBC attributes."""
+
+    def __init__(self, handle):
+        # type: (int) -> None
+        self._handle = handle
+
+        # Here, we are caching the attribute names and enums to work around a driver bug.
+        # The bug results in an empty attribute value after intermixing calls to get attribute values and enums.
+        # We can avoid the bug if we get all attribute enums first, before getting any attribute values.
+        self._cache = {}  # type: typing.Dict[typing.Text, typing.List[typing.Text]]
+        for name in self._get_names():
+            self._cache[name] = self._get_enums(name)
```

Comment:
You could make this more "pythonic" by doing the following.

self._cache =dict(
   (name, self._get_enums(name))
  for name in self._get_names)
)

Reply:
Fixed.